### PR TITLE
GRPC GraknTx Getters

### PR DESCRIPTION
**Pull request by fppt**
# Why is this PR needed?

To expand on the GRPC api functionality

# What does the PR do?

Addes getters for schema concepts

# Does it break backwards compatibility?

No

# List of future improvements not on this PR

More Methods on GRPC

### CODE REVIEW

File: grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java [hunk 1]
```diff
@@ -90,38 +94,40 @@ public GrpcClient client() {
...
 
-    private <X extends Concept> X putSchemaConcept(Label label, Schema.MetaSchema meta, @Nullable Function<VarPattern, VarPattern> extender){
+    private <X extends SchemaConcept> X putSchemaConcept(Label label, Schema.MetaSchema meta,
+                                                   Supplier<X> supplier,
+                                                   @Nullable Function<VarPattern, VarPattern> extender){
         VarPattern var = var().label(label).sub(var().label(meta.getLabel()));
         if(extender != null) var = extender.apply(var);
         queryRunner().run(Graql.define(var));
-        return null;
+        return supplier.get();
```
Comment (aelred) [inline grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java]: Haha, I didn't even notice this was returning `null` before!

Is it possible to just return the answer from the `define` query?

```
return queryRunner.run(Graql.define(var)).get("x").asSchemaConcept();
```

File: grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java [hunk 2]
```diff
@@ -215,32 +234,32 @@ public void commit() throws InvalidKBException {
 
     @Override
     public Type getMetaConcept() {
-        throw new UnsupportedOperationException(); // TODO
+        return getSchemaConcept(Schema.MetaSchema.THING.getLabel());
```
Comment (aelred) [inline grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java]: more default methods? :D

File: grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java [hunk 3]
```diff
@@ -133,13 +139,13 @@ public Role putRole(Label label) {
     @Nullable
     @Override
     public <T extends SchemaConcept> T getSchemaConcept(Label label) {
-        throw new UnsupportedOperationException(); // TODO
+        return getSchemaConcept(label);
```
Comment (aelred) [inline grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java]: giphy

File: grakn-test/test-integration/src/test/java/ai/grakn/test/engine/GrpcServerIT.java [hunk 4]
```diff
@@ -112,6 +112,19 @@ public void whenPuttingEntityType_EnsureItIsAdded(){
...
+    @Test
+    public void whenGettingEntityType_EnsureItIsReturned(){
+        String label = "Oliver";
+        try (GraknTx tx = localSession.open(GraknTxType.WRITE)) {
+            tx.putEntityType(label);
+            tx.commit();
+        }
+
+        try (GraknTx tx = remoteSession.open(GraknTxType.WRITE)){
+            assertNotNull(tx.getEntityType(label));
```
Comment (aelred) [inline grakn-test/test-integration/src/test/java/ai/grakn/test/engine/GrpcServerIT.java]: We could assert the label is correct to be more sure we have the right thing... but I guess that has to wait for my PR #2656

File: grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java [hunk 5]
```diff
@@ -150,31 +156,44 @@ public Role putRole(Label label) {
...
+    }
+
+    @Nullable
+    private <X extends SchemaConcept> X getSchemaConcept(Label label, @Nullable Schema.MetaSchema meta){
+        VarPattern pattern = var().label(label);
+        if(meta != null) pattern = pattern.sub(var().label(meta.getLabel()));
+        Optional<Answer> result = queryRunner().run(Graql.match(pattern).get()).findAny();
+        if(result.isPresent()){
+            Collection<Concept> answer = result.get().concepts();
+            if (!answer.isEmpty()) return (X) Iterables.getOnlyElement(answer);
```
Comment (aelred) [inline grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java]: Maybe this is simpler:
```
Var var = var("x");
VarPattern pattern = var.label(label);
...
if (result.isPresent()) {
     return (X) result.get().get(var);
}
```

File: grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java (hunk 2 above)
Comment (fppt) [reply grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java]: SOLD!

File: grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java (hunk 3 above)
Comment (fppt) [reply grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java]: ya ya PMD already yelled at me :D

**Pull request merged**